lib/prx_auth/rails.rb

```diff
@@ -6,10 +6,36 @@
 module PrxAuth
   module Rails
     class << self
-      attr_accessor :configuration
+      attr_accessor :configuration, :installed_middleware
 
       def configure
-        yield configuration
+        yield configuration if block_given?
+
+        # only install from first call to configure block
+        if configuration.install_middleware && !installed_middleware
+          install_middleware!
+          self.installed_middleware = true
+        end
+      end
+
+      def install_middleware!(app = nil)
+        app ||= ::Rails.application if defined?(::Rails)
```

Comment:
"||" takes precedence over "if" so this should work https://ruby-doc.org/core-2.2.0/doc/syntax/precedence_rdoc.html

Reply:
Ha- I didn't even think of that conflict.  Glad you looked it up!